mantidproject/mantid
# Engineering GUI replace CSV fitting output with HDF5

This PR replaces the CSV output from the Fitting tab of the ED GUI with HDF5 output. A new algorithm, `EnggSaveSinglePeakFittingOutputToHDF5`, was implemented along with all the trimmings, and slotted into the Fitting presenter in place of `SaveDiffFittingAscii`.

`SaveDiffFittingAscii` (**to my knowledge**) is now surplus to requirement, so has been deprecated.

Some dead code relating to 'multi-run mode', which doesn't really exist in the GUI any more, was also removed

**To test:**

Code review, make sure unit tests and documentation pass locally.

Some sample data is [here](https://github.com/mantidproject/mantid/files/1960306/Focus.zip)

Load up the ED GUI (if you've not opened it before, you'll be prompted for an RB number - just put anything in here). Load the three files in to **Focused Run #**, and plot one by clicking on its label in the list next to the plot area. Click **Clear** next to the **Peaks** input, and then click **Select Peak** to enable the peak picker. Select a few peaks on the plot - do this by Shift-clicking on a peak and then clicking **Add Peak**. Click **Fit All**.

Go to your output directory (this will be `C:\EnginX_Mantid\User\<rb_number>\Runs` on Windows and `~/EnginX_Mantid/User/<rb_number>/Runs` on Linux. There should be 2 HDF5 files, one for 277208 and one for 277209. Open them up and make sure that there are results for both banks for 277208, and results for bank 1 of 277209,

*Edit*: there should now also be a merged file, `277208-277209`. Open this up and make sure that the contents of the SinglePeakFitting groups match those of the corresponding individual files

Partially address #21446 
Also closes #17168 and #17206, as any issues related to `SaveDiffFittingAscii` are now irrelevant

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/873cf68e89bfc8e2533400300fbcee565ace7db1)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
